### PR TITLE
ref(eventstore): Move get latest/earliest event to eventstore

### PR DESCRIPTION
**Pull request by lynnagara**
Since this is pretty similar to getting a next/previous event, let's move this responsibility into the Eventstore.

### CODE REVIEW

**Review by markstory — COMMENTED** [review]
Should there be tests for the eventstore methods?

**Comment by lynnagara** [comment]
@markstory Good call, added a test for oldest/latest

File: src/sentry/api/bases/organization_events.py [hunk 1]
```diff
@@ -108,72 +102,3 @@ def get_snuba_query_args_legacy(self, request, organization):
             )
 
         return snuba_args
-
-    def next_event_id(self, snuba_args, event):
```
Comment (fpacifici) [inline src/sentry/api/bases/organization_events.py]: What's the reason for moving these methods into OrganizationEventDetailsEndpoint ? Were they not used anywhere else ?

File: src/sentry/eventstore/base.py [hunk 2]
```diff
@@ -198,6 +200,28 @@ def get_prev_event_id(self, event, filter):
...
+        Gets the earliest event given a current event and some conditions/filters.
+        Returns a tuple of (project_id, event_id)
+
+        Arguments:
+        event (Event): Event object
+        filter (Filter): Filter
+        """
+        raise NotImplementedError
+
+    def get_latest_event_id(self, event, filter):
```
Comment (fpacifici) [inline src/sentry/eventstore/base.py]: nit: naming is a bit inconsistent: 
oldest/newest 
earliest/latest

Comment (lynnagara) [reply src/sentry/eventstore/base.py]: Thanks, updated

File: src/sentry/api/bases/organization_events.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/bases/organization_events.py]: Thanks for doing this cleanup :+1:

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**